### PR TITLE
lldb: Test more invariants in Swift async backtrace test and adapt un…

### PR DESCRIPTION
**Pull request by fredriss**
…winder logic

### CODE REVIEW

**Comment by fredriss** [comment]
@swift-ci test

**Comment by fredriss** [comment]
@swift-ci test

**Comment by fredriss** [comment]
@swift-ci test macOS Platform

File: lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py [hunk 1]
```diff
@@ -11,31 +11,66 @@ class TestSwiftAsyncBacktraceLocals(lldbtest.TestBase):
...
+                fibonacci_number = 10-n+f
+                self.assertIn("fibonacci", frame.GetFunctionName(),
+                              "Redundantly confirm that we're stopped in fibonacci()")
+                if not cfa[fibonacci_number]:
+                    cfa[fibonacci_number] = frame.GetCFA()
+                else:
+                    self.assertEqual(cfa[fibonacci_number], frame.GetCFA(),
+                                     "Stable CFA for the first 10 recursions")
+                # Because we use entry values, we want the async context of a
+                # frame to be presented in the "calling" frame aync context register
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py:55]: presented -> present?

File: lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py [hunk 2]
```diff
@@ -11,31 +11,66 @@ class TestSwiftAsyncBacktraceLocals(lldbtest.TestBase):
...
+                if fibonacci_number > 1 and cfa[fibonacci_number-1]:
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py:56]: why is it `fibonacci_number > 1`?

isn't `cfa[fibonacci_number-1]` guaranteed to exist at this point?

File: lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py [hunk 3]
```diff
@@ -11,31 +11,66 @@ class TestSwiftAsyncBacktraceLocals(lldbtest.TestBase):
...
+                    self.assertEqual(frame.register[async_context_reg].GetValueAsUnsigned(),
+                                     cfa[fibonacci_number-1], "Context register presented in the right frame")
+
+                if f == 0:
+                    # Get arguments (arguments, locals, statics, in_scope_only)
+                    args = frame.GetVariables(True, False, False, True)
+                    self.assertEqual(len(args), 1, "Found one argument")
+                    self.assertEqual(args[0].GetName(), "n", "Found n argument")
+                    self.assertEqual(args[0].GetValue(), str(10-n), "n has correct value")
+                if f != 0:
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py:66]: `else`?

File: lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py [hunk 4]
```diff
@@ -11,31 +11,66 @@ class TestSwiftAsyncBacktraceLocals(lldbtest.TestBase):
...
+                    # The PC of a frame logical is stored in its "callee"
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/async/unwind/backtrace_locals/TestSwiftAsyncBacktraceLocals.py:67]: s/frame logical/logical frame/

**Review by kastiglione — APPROVED** [review]
nice software. A few minor comments

**Pull request closed without merging**